src/address.rs

```diff
@@ -204,4 +255,27 @@ mod tests {
         let ips = parse_addresses(&opts);
         assert_eq!(ips.len(), 0);
     }
+
+    #[test]
+    fn resolver_default_cloudflare() {
+        let opts = Opts::default();
+
+        let resolver = get_resolver(&opts.resolver);
+        let lookup = resolver.lookup_ip("www.example.com.").unwrap();
+
+        assert!(opts.resolver.is_none());
+        assert!(lookup.iter().next().is_some());
+    }
+
+    #[test]
+    fn resolver_args_google_dns() {
+        let mut opts = Opts::default();
+        // https://developers.google.com/speed/public-dns
+        opts.resolver = Some("8.8.8.8,8.8.4.4".to_owned());
+
+        let resolver = get_resolver(&opts.resolver);
+        let lookup = resolver.lookup_ip("www.example.com.").unwrap();
+
+        assert!(lookup.iter().next().is_some());
+    }
```

Comment:
I think we try to avoid network tests so you can test on machines without the internet, in this case mocking the test output 🤔 

But I think for the things which require this (Homebrew requires no networked tests) we have our own different tests specific to it

Although generally I regualrly make the argument that you cannot test a network CLI tool without some kind of network 😂  so approved!

Reply:
Yep, I mulled this one over for a while: the only thing I could think would be to wrap the `Resolver` in a "New Type" which could capture the config. sent to it an expose that. That way, we could:

a. better test that each branch of `get_resolver` results in the right DNS resolvers being referenced.
b. potentially derive a `StructOpt` `from_os_str` for it and create it upstream in `Opts`, isolating a bit more from the parsing code.

…but that seemed a bit much after a hefy `rebase` :grin:

Reply:
I'm aware in PyTest that you can add "tags" to tests, so you could create  tests like "pytest --tag online" to test the online features

that way we can keep it to testing offline for locally and online for CI etc....

Not sure if cargo supports that, and again it isn't really a good solution :( 

Jeeze, testing networking tools is hard!